mlir/lib/IR/BuiltinTypes.cpp

```diff
@@ -646,35 +646,43 @@ LogicalResult MemRefType::verify(function_ref<InFlightDiagnostic()> emitError,
 }
 
 bool MemRefType::areTrailingDimsContiguous(int64_t n) {
-  if (!isLastDimUnitStride())
-    return false;
+  return getMaxContiguousTrailingDims() >= std::min(n, getRank());
```

Comment:
```suggestion
  assert(n <= getRank() && "doesn't have this many trailing dims to check for contiguity");
  return getMaxContiguousTrailingDims() >= n;
```

Maybe? Trying to understand the logic for the case where n > getRank().

Reply:
Nope, no logic here, an assertion is more appropriate.